core/internal/tensorboard/tfeventconverter.go

```diff
@@ -23,146 +23,70 @@ type TFEventConverter struct {
 	pluginNameByTag map[string]string
 }
 
-// tagAndJSON is a tag and a JSON representation of a value.
-type tagAndJSON struct {
-	// tag is a '/'-separated key for the metric.
-	tag string
-
-	// json is a JSON-encoded value for the metric that's understood by W&B.
-	json string
-}
-
-// ConvertNext returns a W&B history record corresponding to a TF event.
+// ConvertNext adds data from a TF event to the run.
 //
 // This should be called on events in the order they are read from
 // tfevents files.
-//
-// Returns nil if there's no relevant history data in the event.
-// Errors are logged via the logger and the corresponding data is ignored.
 func (h *TFEventConverter) ConvertNext(
+	emitter Emitter,
 	event *tbproto.TFEvent,
 	logger *observability.CoreLogger,
-) *spb.PartialHistoryRequest {
-	// Maps slash-separated tags to JSON values.
-	jsonData := make([]tagAndJSON, 0, len(event.GetSummary().GetValue()))
+) {
+	stepKey, err := h.withNamespace("global_step")
```

Comment:
could you leave a comment about what special with global_step

Reply:
Unfortunately nothing is special---I was confused before because I messed with my workspace for the "uncategorized" project. I think "global_step" is the suggestion we give to users in some documentation.

I would prefer "tensorboard_step", but "global_step" is consistent with what we do in previous versions.